src/libraries/Microsoft.PowerFx.Core/Public/Marshal/IdentityMarshalerProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// Marshal dotnet types that already are FormulaValues. 
+    /// </summary>
+    public class IdentityMarshalerProvider : ITypeMashalerProvider
+    {
+        public ITypeMarshaler New(Type type, TypeMarshallerCache cache, int maxDepth)
+        {
+            if (typeof(FormulaValue).IsAssignableFrom(type))
+            {
+                return new IdentityMarshaler
+                {
+                    Type = null // $$$ Fix this 
```

Comment:
>Type = null // $$$ Fix this  [](http://example.com/codeflow?start=20&length=28)

this may be another switch case.  Similar to what we do in ForumulaValue.New